doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
-updates in upstream projects at our pace.
+    $ pytest yt/visualization/tests/test_plotwindow.py::test_all_fields
 
-If you would like to add a new dependency for yt (even an optional dependency)
-or would like to update a version of a yt dependency, you must edit the
-``tests/test_requirements.txt`` file, this path is relative to the root of the
-repository. This file contains an enumerated list of direct dependencies and
-pinned version numbers. For new dependencies, simply append the name of the new
-dependency to the end of the file, along with a pin to the latest version
-number of the package. To update a package's version, simply update the version
-number in the entry for that package.
+See this `link <https://docs.pytest.org/en/stable/usage.html?highlight=invocation>`_ for more on how to invoke pytest.
 
-Finally, we also run a set of tests with "minimal" dependencies installed. Please make sure any new tests you add that depend on an optional dependency are properly set up so that the test is not run if the dependency is not installed. If for some reason you need to update the listing of packages that are installed for the "minimal" dependency tests, you will need to edit ``tests/test_minimal_requirements.txt``.
 
-How to Write Unit Tests
-^^^^^^^^^^^^^^^^^^^^^^^
+Unit Test Tools
+^^^^^^^^^^^^^^^
 
-yt provides several pieces of testing assistance, all in the ``yt.testing``
-module.  Describing them in detail is somewhat outside the scope of this
-document, as in some cases they belong to other packages.  However, a few come
+yt provides several tools to assist with writing unit tests. These tools all reside in the ``yt.testing``
+module.  Describing them all in detail is somewhat outside the scope of this
+document. However, a few come
 in handy:
 
 * :func:`~yt.testing.fake_random_ds` provides the ability to create a random
   dataset, with several fields and divided into several different
-  grids, that can be operated on.
+  grids.
 * :func:`~yt.testing.assert_equal` can operate on arrays.
 * :func:`~yt.testing.assert_almost_equal` can operate on arrays and accepts a
   relative allowable difference.
 * :func:`~yt.testing.assert_allclose_units` raises an error if two arrays are
   not equal up to a desired absolute or relative tolerance. This wraps numpy's
-  assert_allclose to correctly verify unit consistency as well.
+  :py:func:`numpy.testing.assert_allclose` to correctly verify unit consistency as well.
 * :func:`~yt.testing.amrspace` provides the ability to create AMR grid
   structures.
 * :func:`~yt.testing.expand_keywords` provides the ability to iterate over
   many values for keywords.
 
+How To Write New Unit Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
 To create new unit tests:
 
 #. Create a new ``tests/`` directory next to the file containing the
-   functionality you want to test and add an empty ``__init__.py`` file to
+   machinery you want to test and add an empty ``__init__.py`` file to
    it.
```

Comment:
Add a sentence here saying what to do if a tests/ folder already exists, since it is likely (unless they are writing a frontend) that there will be one there already.